### ISSUE TITLE
[Bug] ServerLocation url attribute doesn't add the current url to the Router context

### ISSUE THREAD

**Issue by ricardobrandao** [reach/router#108]
Hi,

I'm having an issue while passing the current route to the Router using the `ServerLocation` component within my express server. I'm rendering the following html on the server:
```
const html = ReactDOMServer.renderToString(
  <ServerLocation url={req.url}>
    <App />
  </ServerLocation>
);
```

However, although the `req.url` main contain a different path (e.g. '/contacts'), the Router's current route always fallback to `'/'`.

**Comment by LeeCheneler** [comment]
I believe I'm getting this issue too. Routes don't evaluate on the server 🤔

**Comment by ricardobrandao** [comment]
I believe they do (I've placed a <Location> inside the server-side rendered app), however, as soon as the client-side kicks in, the current url is lost. Unfortunately, I had to move back to react-router since I had to deliver the product. Other than this, the @reach/router is working great and I would be keen to give it another shoot.

**Comment by LeeCheneler** [comment]
Yeah, we POC'd this as we'd *love* to use it. But the server doesn't render the routes, only `/` 😞

**Comment by strothj** [comment]
Some kind of weird race condition is happening. If I disable any redirects in the client code, things seem to work.

As soon as I have a route with a redirect, the server redirects normally the first time. On subsequent renders, the client is stuck with the / route.

**Comment by good-idea** [comment]
I'm having an issue similar to @strothj:

My routes look like:

```
<Router>
	<Homepage path="/" />
	<Project path="/projects/:projectSlug" />
	<Blog path="/blog" />
	<Article path="/blog/:pageSlug" />
	<NotFound default />
</Router>
```

1. I visit `/` and it loads correctly
2. I visit `/blog` and it loads correctly
3. I visit `/blog/some-article` and it loads correctly
4. I visit `/` and it renders `<NotFound />`
5. I visit `/blog` and it renders `<NotFound />`
6. I visit `/blog/another-article` and it loads correctly
7. I visit `/projects/project-one` and it renders `<NotFound />`

It seems like once I load a route with a parameter, something goes funny for all other routes.

**Comment by 4074** [comment]
I'm having this issue too on 1.2.1. And have to back to react-router.

**Comment by peterbe** [comment]
For what it's worth, I *thought* I had this bug too. Turns out, I hadn't added `yarn add @react/router` to the package. It existed in `../node_modules` but not in `./node_modules`. 

I don't have an Express server but I have a `cli` that generates static HTML based on importing `../client/src/App.js` and I forgot to add `@reach/router` in both places. It was a kinda mess but once I got it installed correctly in both places, it now works. Before, that it didn't work. Seems like an error got swallowed incorrectly somewhere. 

Sorry that I can't explain it better but I was getting the same problem until I got my packages in order.

**Comment by davemanton** [comment]
I have this problem at the moment.

I have App as follows 
`
const App = () => (
    <Router>
        <Home path='/' />
        <OtherPage path='/otherpage' />
    </Router>
)
`

With a server version wrapping it as follows 
`
const ServerApp = ({ path }) => (
    <ServerLocation url={path}>
        <App />
    </ServerLocation>
)
`

If I render the client side with no server at all then the routing works perfectly.
If I render the server side with no client it just always defaults to the home page output.

One workaround I have found is to create a location object as it does in the source for the LocationContext and pass that location directly to the router on the server.  This does work.

It seems that on the server the location context isn't actually passing the location object to the router correctly.

**Comment by LeeCheneler** [comment]
I wouldn't adopt this library as Ryan has gone back to ReactTraining and they're going to roll some of this libraries features into react-router. @reach/router won't be getting anymore active dev work. See this blog post for more details - https://reacttraining.com/blog/reach-react-router-future/

**Comment by davemanton** [comment]
@LeeCheneler thanks for that! Much appreciated heads up :)
